apis/v1alpha1/httproute_types.go

```diff
@@ -560,6 +574,65 @@ type HTTPRequestMirrorFilter struct {
 	Port *PortNumber `json:"port,omitempty"`
 }
 
+// HTTPRequestAuthorization defines configuration for the RequestAuthorization filter.
+type HTTPRequestAuthorization struct {
+	// URL specifies the address of the service used to perform the authorization check.
+	//
+	// Support: Extended
+	//
+	URL string `json:"url"`
+
+	// Protocol is used to define the Protocol that the gateway will use to communicate with the authorization
+	// service. If not set, it will default to "http" or inferred from the scheme in URL.
+	// Examples: "grpc","http2"
+	//
+	// Support: Extended
+	//
+	// +optional
+	// +kubebuilder:default=http
+	Protocol string `json:"protocol,omitempty"`
+
+	// FailModeAllow defines how this authorization filter should be handled if it timeouts or fails to
+	// contact the URL defined. This field accepts two options:
+	//
+	// - True: Fail open. Traffic will be allowed even if URL can't be contacted or timeouts
+	// - False: Fail closed. Default value. Traffic will be denied if the gateway can't reach the URL or timeouts.
+	//
+	// Support: Extended
+	//
+	// +optional
+	// +kubebuilder:default=false
+	FailModeAllow *bool `json:"failModeAllow,omitempty"`
+
+	// RequestBody allows to define if the body of the request should be sent to the authorization service.
+	//
+	// - True: The request body will be sent to the authorization endpoint.
+	// - False: Don't send the request body. Default value.
+	//
+	// Support: Extended
+	//
+	// +optional
+	// +kubebuilder:default=false
+	RequestBody *bool `json:"requestBody,omitempty"`
+
+	// TimeoutMs is the duration that the authorization call is allowed to operate.
+	// Default if unset is 100
+	//
+	// Support: Extended
+	//
+	// +optional
+	// +kubebuilder:default=100
+	TimeoutMs int `json:"timeoutMs,omitempty"`
+
+	// Metadata can be used to provide additional context information for the authorization
+	// service to make a decision.
+	//
+	// Support: Extended
+	//
+	// +optional
+	Metadata map[string]string `json:"metadata,omitempty"`
```

Comment:
AFAIK, only the envoy GRPC protocol can pass arbitrary key value pairs here. How do you indicate missing or partial support for this field?

Reply:
I think this is for the implementor to decide; for example, in Nginx, this could be implemented as a custom HTTP header sent to the authorization service. Perhaps we should put some validation in place to make that easier, like a max length and allowed characters.

If the implementation doesn't support this, it can reject the route and set the proper status condition.